pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -799,6 +813,26 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	return nil
 }
 
+func (c *controller) restartFailedPopulator(ctx context.Context, pvc *corev1.PersistentVolumeClaim, namespace, podName string) (err error) {
+	err = c.updatePvc(ctx, pvc, namespace)
```

Comment:
I'd separate this to a separate function - it's not part of restarting the populator pod

Reply:
if you say that this methods is responsible for triggering another attempt, including updating the counter, that's fine but then let's make it do it (increment the counter)